compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -861,68 +960,60 @@ execMigrate projectOpts inFile1_ inFile2_ mbDir =
       withProjectRoot' projectOpts $ \_relativize
        -> do
           -- get the package name and the lf-package
-          [(pkgName1, pkgId1, lfPkg1, depMap1), (pkgName2, pkgId2, lfPkg2, depMap2)] <-
+          [(pkgName1, _pkgId1, lfPkg1, sdkVer1), (pkgName2, _pkgId2, lfPkg2, sdkVer2)] <-
               forM [inFile1, inFile2] $ \inFile -> do
                   bytes <- B.readFile inFile
                   let dar = ZipArchive.toArchive $ BSL.fromStrict bytes
                   -- get the main pkg
                   dalfManifest <- either fail pure $ readDalfManifest dar
                   let pkgName = takeBaseName $ mainDalfPath dalfManifest
-                  let dalfs = dalfPaths dalfManifest
-                  deps <-
-                      forM dalfs $ \dalfPath -> do
-                          dalf <- getEntry dalfPath dar
-                          (pkgId, _pkg) <- decode $ BSL.toStrict $ ZipArchive.fromEntry dalf
-                          pure (pkgId, stringToUnitId $ takeBaseName dalfPath)
                   mainDalfEntry <- getEntry (mainDalfPath dalfManifest) dar
                   (mainPkgId, mainLfPkg) <- decode $ BSL.toStrict $ ZipArchive.fromEntry mainDalfEntry
-                  pure (pkgName, mainPkgId, mainLfPkg, MS.fromList deps)
-          let pkgMap = MS.union depMap1 depMap2
+                  pure (pkgName, mainPkgId, mainLfPkg, sdkVersion dalfManifest)
           -- generate upgrade modules and instances modules
           let eqModNames =
                   (NM.names $ LF.packageModules lfPkg1) `intersect`
                   (NM.names $ LF.packageModules lfPkg2)
+          thisSdkVer <- getSdkVersion
+          let differentSdks = (length $ nubSort [sdkVer1, sdkVer2, thisSdkVer]) /= 1
```

Comment:
I was thinking of comparing the package id of `daml-stdlib` not the main dalf.

Reply:
ah I see. so you would want to get the version of the `daml-stdlib` to figure out this sdk version or also of all dependencies (which would fail in the rare case that a a package doesn't depend on the daml-stdlib).